uber/site_sections/preregistration.py

```diff
@@ -67,7 +67,11 @@ def unpaid_preregs(self):
     def paid_preregs(self):
         return cherrypy.session.setdefault('paid_preregs', [])
 
```

Comment:
ok yea this is very non-obvious and I only caught it in the debugger.  the issue I'm fixing is real, though, there might be a more pythonic way to deal with this.

# how HTTPRedirect works:
deep down in cherrypy's HTTPRedirect's __init__, it attempts to take our relative URL and create an absolute URL from it by looking at the request data from the browser.

i.e. if you accessed cherrypy with a request string from your browser like 'http://uberserver.com:765/uber/preregistration/someplace' and you ask to redirect to 'index', cherrypy will mix those two things together and generate: 'http://uberserver.com:765/uber/preregistration/index'

likewise, if the way you accessed ubersystem was different (let's say you visited the page through another proxy server somehow) it still generates the correct result.  

If you later accessed this same ubersystem instance via 'http://localhost8765/uber/preregistration/something' and redirected to 'coolestpageever', cherrypy will look at your request string and build the following URL: 'http://localhost8765/uber/preregistration/coolestpageever'

TLDR: Cherrypy creates a base URL by looking at the query string your browser sent it.

So, HTTPRedirect() only does that generation on __init__() when it's created, and it needs to have the context i.e. access to THE CURRENT request string from the browser.

# Why the current implementation might be an issue

This was really non-obvious and I only caught it by accident in the debugger.

When we did had this function definition:

```python
def _get_unsaved(self, id, if_not_found=HTTPRedirect('index')):
```

The *definition* was creating a HTTPRedirect object ```at import time``` (i.e. at sideboard start). 

This HTTPRedirect instance would not have any request context to look at (since this code is not being executed as part of filling a browser request)

Cherrypy doesn't crash on this, but instead of building an absolute path using the querystring, it just kind of guesses based on config data.  The comments even say this might cause issues:
```python
# No request.app (we're being called outside a request).
# We'll have to guess the base from server.* attributes.
# This will produce very different results from the above
# if you're using vhosts or tools.proxy.
```

(that code is in __init__.py in cherrypy) in the function ```def url(path="", qs="", script_name=None, base=None, relative=None):```

# short version

Putting HTTPRedirect() in the function definition makes a global object that we can later 'raise' if needed.  The problem is it will not look at a querystring, and thus, might fail in various configurations of proxy stuff.

My goal was to make it so that no matter how your browser ended up getting to cherrypy's port, you get the right data back out of it with minimal tricks.

So my fix here is simply: We shouldn't create HTTPRedirect objects and save them in a global state (which is what the original code did), we should only create them right before they're needed.